### DISCUSSION TITLE
Securing Open Ai Api key

### DISCUSSION THREAD

**Discussion by powercore2000** [KellanM/OpenAI-Api-Unreal#22] [General]
How secure would it be to have the API key set as an environment variable in an unreal project? Assuming I use the Encrypt UAsset Files option, could someone still retrieve the decryption key from memory, and find the Api key inside the project files?

**Comment by therestartprince** [comment]
- ### Is it possible to create and encryption container
- ### <sub>(A virtual Volume, not sure there might be way better options)</sub>
- ### For the API key and attach strict permissions on its use and stops that prevent it from permanantly
- ### being## imprinted on the source code?

1. ### That way you stop general leakage
2. ### If someone were actively trying to steal it,
3. ### They would have to know exactly when it was being accessed and used to even attempt to take it.
4. ### I cant think of a better option off the top of my head

- [ ] I want to do something like this for GODOT
- [ ] If it is not being done already
